libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
Don't have strong opinions either way, would be nice to be compatible with slightly older versions of Chef if we can, but I'm hoping most people on a systemd distro are keeping up to date with Chef as well.